diff -urN mysql-5.7.20.orig/storage/innobase/row/row0sel.cc mysql-5.7.20/storage/innobase/row/row0sel.cc
--- mysql-5.7.20.orig/storage/innobase/row/row0sel.cc	2017-11-09 15:40:55.000000000 +0000
+++ mysql-5.7.20/storage/innobase/row/row0sel.cc	2017-11-09 16:10:48.000000000 +0000
@@ -5193,7 +5193,7 @@
 		reporting due to search views etc. */
 		if (prev_rec != NULL
 		    && prebuilt->m_mysql_handler->end_range != NULL
-		    && prebuilt->idx_cond == false && end_loop >= 100) {
+		    && prebuilt->idx_cond == NULL && end_loop >= 100) {
 
 			dict_index_t*	key_index = prebuilt->index;
 			bool		clust_templ_for_sec = false;
