From de78a18e0dbe810eaafa3366ec03f54ad6c19792 Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Mon, 19 Oct 2020 23:48:16 -0400
Subject: [PATCH 04/23] Add include header for std::string

---
 sql/create_def.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/sql/create_def.cc b/sql/create_def.cc
index cb99224dc4c..749cd507151 100644
--- a/sql/create_def.cc
+++ b/sql/create_def.cc
@@ -69,6 +69,7 @@
 #include <thread>
 #include <unordered_set>
 #include <vector>
+#include <string>
 
 /** Prints an error message supplied and attaches GetLastError with formatted
  * message. */
-- 
2.48.1

