From f8b131862392d1df10e0b129e41be3530d2ffb46 Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Thu, 23 Jul 2020 12:56:13 -0400
Subject: [PATCH 05/23] Remove forceful injection of debug info

---
 cmake/os/Windows.cmake | 17 +++++++----------
 1 file changed, 7 insertions(+), 10 deletions(-)

diff --git a/cmake/os/Windows.cmake b/cmake/os/Windows.cmake
index b9428da34a4..f0f98aa56f9 100644
--- a/cmake/os/Windows.cmake
+++ b/cmake/os/Windows.cmake
@@ -162,13 +162,13 @@ IF(MSVC)
   # - Disable optimization
   #     /Od
   #     Turns off all optimizations in the program and speeds compilation.
-  FOREACH(lang C CXX)
-    IF(FPROFILE_GENERATE OR FPROFILE_USE)
-      SET(CMAKE_${lang}_FLAGS_RELEASE "${CMAKE_${lang}_FLAGS_RELEASE} /Zi")
-    ELSE()
-      SET(CMAKE_${lang}_FLAGS_RELEASE "${CMAKE_${lang}_FLAGS_RELEASE} /Z7")
-    ENDIF()
-  ENDFOREACH()
+  #  FOREACH(lang C CXX)
+  #    IF(FPROFILE_GENERATE OR FPROFILE_USE)
+  #      SET(CMAKE_${lang}_FLAGS_RELEASE "${CMAKE_${lang}_FLAGS_RELEASE} /Zi")
+  #    ELSE()
+  #      SET(CMAKE_${lang}_FLAGS_RELEASE "${CMAKE_${lang}_FLAGS_RELEASE} /Z7")
+  #    ENDIF()
+  #  ENDFOREACH()
 
   # Defaults initialized by cmake:
   # DEBUG          /MDd /Zi /Od  /Ob0 /RTC1
@@ -187,9 +187,6 @@ IF(MSVC)
     ENDIF()
     IF(FPROFILE_GENERATE OR FPROFILE_USE)
       STRING(REPLACE "/ZI"  "/Zi" "${flag}" "${${flag}}")
-    ELSE()
-      STRING(REPLACE "/Zi"  "/Z7" "${flag}" "${${flag}}")
-      STRING(REPLACE "/ZI"  "/Z7" "${flag}" "${${flag}}")
     ENDIF()
     IF (NOT WIN_DEBUG_NO_INLINE)
       STRING(REPLACE "/Ob0"  "/Ob1" "${flag}" "${${flag}}")
-- 
2.48.1

