From de8afa8237d3e67de256b391783be8b197dd65ed Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Mon, 19 Jun 2023 20:25:57 +0100
Subject: [PATCH 18/23] Make libfido checks conditional

---
 CMakeLists.txt | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9c15bcea1a6..5592c23c85b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2033,8 +2033,10 @@ ENDIF()
 MYSQL_CHECK_RAPIDJSON()
 
 # Look for fido2 library
-MYSQL_CHECK_FIDO()
-MYSQL_CHECK_FIDO_DLLS()
+IF(WITH_AUTHENTICATION_CLIENT_PLUGINS)
+  MYSQL_CHECK_FIDO()
+  MYSQL_CHECK_FIDO_DLLS()
+ENDIF()
 
 IF(APPLE)
   GET_FILENAME_COMPONENT(HOMEBREW_BASE ${HOMEBREW_HOME} DIRECTORY)
-- 
2.48.1

