From 6e90541ef905516f5a414943c892f0ee3fa577fe Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Tue, 1 Apr 2025 23:14:29 +0100
Subject: [PATCH 23/23] Allow setting WITH_SASL=none

---
 cmake/sasl.cmake | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/cmake/sasl.cmake b/cmake/sasl.cmake
index 8c4fa22340e..7b5e6cc7df1 100644
--- a/cmake/sasl.cmake
+++ b/cmake/sasl.cmake
@@ -270,6 +270,9 @@ MACRO(MYSQL_CHECK_SASL)
     ELSE()
       MESSAGE(FATAL_ERROR "-DWITH_SASL=<path> not supported on this platform")
     ENDIF()
+  ELSEIF(WITH_SASL STREQUAL "none")
+    SET(SASL_FOUND FALSE)
+    MESSAGE(STATUS "SASL path is none, disabling sasl support.")
   ELSE()
     RESET_SASL_VARIABLES()
     IF(NOT WIN32)
-- 
2.48.1

